bondcpp/CMakeLists.txt

```diff
@@ -5,9 +5,12 @@ if(NOT CMAKE_CXX_STANDARD)
   set(CMAKE_CXX_STANDARD 14)
 endif()
 
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
```

Comment:
While this probably does make it work on Windows, this is not the style that we have used in all of the rest of our packages.  In short, we explicitly add a `visibility_control.h` file to each of our projects, and then add the `RCLCPP_PUBLIC` (or whatever) on the symbols that need to be visible.  You can see an example of this at https://github.com/ros2/rcpputils/blob/master/include/rcpputils/visibility_control.hpp.  I'll suggest that we do the same here.

Reply:
Ack, @Tobias-Fischer let me know if you prefer to do that or if you prefer that I do it, thanks!

Reply:
It would be great if you could do it @traversaro :)

Reply:
Done in https://github.com/ros/bond_core/pull/83/commits/ea45bfbb4f0d0fe4bfa206793c9112664e531f98 . I also fixed `test_bond` compilation to make sure the symbols were correctly exported.

Reply:
I realized I never mentioned @clalancette once I did this, so probably he missed this.